vector/src/main/java/im/vector/activity/VectorHomeActivity.java

```diff
@@ -774,6 +779,47 @@ private void markAllMessagesAsRead() {
     }
 
     /**
+     * Send a read receipt when the device is offline, for each room, on the last
+     * message of the room.
+     * @return true if operation was performed, false otherwise
+     */
+    private boolean markAllMessagesAsReadWhenOffline() {
+        boolean isOperationDone = false;
+
+        if(!Matrix.getInstance(this).isConnected()) {
+
+            ArrayList<MXSession> sessionsList = new ArrayList<>(Matrix.getMXSessions(this));
+            mReadReceiptSessionListIterator = sessionsList.iterator();
+
+            if (null == sessionsList) {
+                Log.w(LOG_TAG, "## markAllMessagesAsReadWhenOffline(): invalid session list");
+            } else {
+                for (MXSession session : sessionsList) {
+                    if (null != session) {
+                        // test if the session is still alive i.e the account has not been logged out
+                        ArrayList<Room> roomCompleteList = new ArrayList<>(session.getDataHandler().getStore().getRooms());
+
+                        if(null != roomCompleteList) {
+                            // for each room send the receipt for the latest message
+                            for (Room room : roomCompleteList) {
+                                room.sendReadReceipt(null);
+                                isOperationDone = true;
```

Comment:
isOperationDone |= room.sendReadReceipt(null);
seems better : it is possible there is no update. sendReadReceipt returns true when there is an update.